sechub-test/src/test/java/com/daimler/sechub/test/RoutesTest.java

```diff
@@ -0,0 +1,174 @@
+package com.daimler.sechub.test;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.security.RolesAllowed;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.daimler.sechub.sharedkernel.RoleConstants;
+
+import static org.junit.Assert.*;
+
+@SuppressWarnings("rawtypes")
+public class RoutesTest {
+	
+	private Map<Class, Map<String, ArrayList<String>>> classRoutesRolesMap = new HashMap<Class, Map<String, ArrayList<String>>>();
+	
+	@Before
+	public void before() {
+		List<Class> allClasses = getAllClassesForAnnotationType(RestController.class);
+		
+		for (Class clazz : allClasses) {
+			getClassUserRolePaths(clazz);
+		}		
+	}
+	
+	@Test
+	public void routes_assigned_to_right_roles() {
+			
+		for (Entry<Class, Map<String, ArrayList<String>>> entry : classRoutesRolesMap.entrySet()) {
+			
+			String clazzName = entry.getKey().getName();
+			Map<String, ArrayList<String>> routesEntries = entry.getValue();
+			
+			for (Entry<String, ArrayList<String>> routesEntry : routesEntries.entrySet()) {
+				String route = routesEntry.getKey();
+				String[] roles = routesEntry.getValue().toArray(new String[routesEntry.getValue().size()]);
+				
+				if (!(route.contains("anonymous") || route.contains("actuator"))) {
```

Comment:
What about "/anonymous/" and "/actuator/" ?

If there is anybody designing a rest part containing those words in another context... yes I know a little bit fictional, but we should be aware

Reply:
done